mainhandler/vulnscan.go

```diff
@@ -283,42 +285,62 @@ func (actionHandler *ActionHandler) scanWorkload(ctx context.Context, sessionObj
 	if err != nil {
 		return fmt.Errorf("failed to get workload %s with err %v", actionHandler.wlid, err)
 	}
-	pod := actionHandler.getPodByWLID(workload)
+
+	pod := actionHandler.getPodByWLID(ctx, workload)
 	if pod == nil {
 		logger.L().Info(fmt.Sprintf("workload %s has no podSpec, skipping", actionHandler.wlid))
 		return nil
 	}
+	mapContainerToImageID := make(map[string]string) // map of container name to image ID. Container name is unique per pod
+
+	// look for container to imageID map in the command args. If not found, look for it in the wl. If not found, get it from the pod
+	if val, ok := actionHandler.command.Args[containerToImageIdsArg].(map[string]string); !ok {
+		// get from pod
+		if len(pod.Status.ContainerStatuses) == 0 {
+			mapContainerToImageID, err = actionHandler.getContainerToImageIDsMap(workload)
+			if err != nil {
+				logger.L().Ctx(ctx).Error(fmt.Sprintf("failed to get container to image ID map for workload %s with err %v", actionHandler.wlid, err))
+			}
+		} else {
+			// get from wl
+			for contIdx := range pod.Status.ContainerStatuses {
+				imageID := pod.Status.ContainerStatuses[contIdx].ImageID
+				mapContainerToImageID[pod.Status.ContainerStatuses[contIdx].Name] = watcher.GetImageID(imageID)
+			}
+		}
+	} else {
+		// get from args
+		mapContainerToImageID = val
+	}
+
 	// get all images of workload
 	errs := ""
 	containers, err := listWorkloadImages(workload)
 	if err != nil {
 		return fmt.Errorf("failed to get workloads from k8s, wlid: %s, reason: %s", actionHandler.wlid, err.Error())
 	}
 
-	// we want running pod in order to have the image hash
-	// actionHandler.getRunningPodDescription(pod)
-
 	for i := range containers {
-
+		imgID := ""
+		if val, ok := mapContainerToImageID[containers[i].container]; !ok {
+			logger.L().Ctx(ctx).Error(fmt.Sprintf("failed to get image ID for container %s", containers[i].container))
```

Comment:
we can continue with the loop if we didn't find the image hash (even though it shouldn't happen)